### PR TITLE
[globalize] Add dateToPartsFormatter from 1.5.0

### PR DESCRIPTION
**Pull request by dantman**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/globalizejs/globalize/releases/tag/1.5.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/globalize/dist/globalize/date.d.ts [hunk 1]
```diff
@@ -49,11 +54,22 @@ declare module "../globalize" {
          */
         dateFormatter(options?: DateFormatterOptions): (value: Date) => string;
 
+        /**
+         * .dateToPartsFormatter( options )
+         * @param {DateFormatterOptions} options see date/expand_pattern for more info.
+         * * @returns {Function} Return a function that formats a date into parts tokens according to the given options. The default formatting is numeric year, month, and day (i.e., `{ skeleton: "yMd" }`).
```
Comment (amcasey) [inline types/globalize/dist/globalize/date.d.ts:60]: Extra `*`?

File: types/globalize/dist/globalize/date.d.ts [hunk 2]
```diff
@@ -32,6 +32,11 @@ declare module "../globalize" {
          */
         timeZone?: string;
     }
+    type DateFormatPartTypes = "day" | "dayperiod" | "era" | "hour" | "literal" | "minute" | "month" | "second" | "zone" | "weekday" | "year";
```
Comment (amcasey) [inline types/globalize/dist/globalize/date.d.ts:35]: Out of curiosity, what's the `literal` part?

**Review by amcasey — CHANGES_REQUESTED** [review]
Minor questions - happy to merge as-is.

**Review by amcasey — APPROVED** [review]

File: types/globalize/dist/globalize/date.d.ts (hunk 2 above)
Comment (dantman) [reply types/globalize/dist/globalize/date.d.ts:35]: Any text in the formatted date that is not part of the date. e.g. Separators.

```js
> Globalize.locale('en-CA')
> Globalize.formatDateToParts(new Date, {datetime: 'full'});
[
  { type: 'weekday', value: 'Tuesday' },
  { type: 'literal', value: ', ' },
  { type: 'month', value: 'May' },
  { type: 'literal', value: ' ' },
  { type: 'day', value: '26' },
  { type: 'literal', value: ', ' },
  { type: 'year', value: '2020' },
  { type: 'literal', value: ' at ' },
  { type: 'hour', value: '6' },
  { type: 'literal', value: ':' },
  { type: 'minute', value: '48' },
  { type: 'literal', value: ':' },
  { type: 'second', value: '29' },
  { type: 'literal', value: ' ' },
  { type: 'dayperiod', value: 'p.m.' },
  { type: 'literal', value: ' ' },
  { type: 'zone', value: 'GMT-07:00' }
]
```

Some locales are also *"fun"* in including a literal "h" character in their time string.
```js
> Globalize.locale('fr-CA');
> Globalize.formatDateToParts(new Date, {time: 'short'});
[
  { type: 'hour', value: '18' },
  { type: 'literal', value: ' h ' },
  { type: 'minute', value: '49' }
]
```

**Comment by dantman** [comment]
Ready to merge

**Pull request merged**